### PR TITLE
syncer: fix errors at some circumstance

### PR DESCRIPTION
**Pull request by andelf**
- skip more unsupported
- alter table ddl split

### CODE REVIEW

**Comment by andelf** [comment]
ready to be reviewed

File: syncer/ast.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+	"strings"
+
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// FIXME: tidb's AST is error-some to handle more condition
+func columnDefToSQL(colDef *ast.ColumnDef) string {
+	typeDef := colDef.Tp.String()
+	sql := fmt.Sprintf("%s %s", columnNameToSQL(colDef.Name), typeDef)
```
Comment (GregoryIan) [inline syncer/ast.go]: use bytes.Buffer and Fprintf is better
```
func genKVs(columns []*column) string {
	var kvs bytes.Buffer
	for i := range columns {
		if i == len(columns)-1 {
			fmt.Fprintf(&kvs, "`%s` = ?", columns[i].name)
		} else {
			fmt.Fprintf(&kvs, "`%s` = ?, ", columns[i].name)
		}
	}

	return kvs.String()
}
```

File: syncer/ast.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	for _, opt := range colDef.Options {
+		switch opt.Tp {
+		case ast.ColumnOptionNotNull:
+			sql += " NOT NULL"
+		case ast.ColumnOptionNull:
+			sql += " NULL"
+		case ast.ColumnOptionDefaultValue:
```
Comment (GregoryIan) [inline syncer/ast.go]: DefaultValue is very common， it's not better to panic

Comment (andelf) [inline syncer/ast.go]: will fix

Comment (andelf) [inline syncer/ast.go]: in order implement this, ExprNode should be walk into, which is complex.
leave it a TODO now. :(

**Comment by andelf** [comment]
@nolouch @iamxy PTAL

**Comment by iamxy** [comment]
merge master first plz

LGTM

**Comment by andelf** [comment]
will try resolve conflicts.
restore modification to sql filter logic or not?

**Comment by GregoryIan** [comment]
add them to sql slice

**Pull request merged**